encode.go

```diff
@@ -37,6 +37,16 @@ func NewEncoder(w http.ResponseWriter) *Encoder {
 	}
 }
 
+// NewCustomEncoder returns a new Encoder to write to the ResponseWriter
+// This allows you to provide a custom http encoder if needed.
+// The encoder must be setup write to the ResponseWriter that is passed in.
+func NewCustomEncoder(w http.ResponseWriter, encoder HTTPEncoder) *Encoder {
+	return &Encoder{
+		encoder: encoder,
+		w:       w,
+	}
+}
+
```

Comment:
I am torn on adding this constructor... While it does allow the use of a different encoder, it feels awkward to use and makes me question if we should do this.

I wonder if we just wait for the need to arise before we add the extra API Surface?

Reply:
Maybe something like this would be better if we wanted to do this.
```
func NewCustomEncoder(w http.ResponseWriter, encoder func(w io.Writer) HTTPEncoder) *Encoder {
	return &Encoder{
		encoder: encoder(w),
		w:       w,
	}
}
```

Reply:
I am on board with just not exposing anything at this point